src/restricted-endpoints/document/document.controller.ts

```diff
@@ -40,6 +43,37 @@ export class DocumentController {
     private permissionService: PermissionService,
   ) {}
 
+  /**
+   * Fetch a document from a database with basic auth.
+   * See {@link https://docs.couchdb.org/en/stable/api/document/common.html?highlight=put%20document#get--db-docid}
+   * @param req express Request object
+   * @param db the name of the database from which the document should be fetched
+   * @param docId the name of the document
+   * @param user logged in user
+   * @param queryParams additional params that will be forwarded
+   */
+  @Head()
+  async headDocument(
+    @Request() req: Req,
+    @Param('db') db: string,
+    @Param('docId') docId: string,
+    @User() user: UserInfo,
+    @Query() queryParams?: any,
+  ): Promise<DatabaseDocument> {
+    const userAbility = this.permissionService.getAbilityFor(user);
+    const document = await firstValueFrom(
+      this.couchdbService.get(db, docId, queryParams),
+    );
+    if (userAbility.can('read', document)) {
+      if (document._rev) {
+        req.res.setHeader('etag', document._rev);
+      }
+      return document;
```

Comment:
according to specs this shouldn't return a document, only the headers, right?

Of course, in order to check permissions we do need to load the doc anyway ... But I would tend to stick to our approach and implement the actual CouchDB API?